src/modules/flow/http-client/http-client.c

```diff
@@ -358,7 +366,17 @@ common_request_finished(void *data,
     SOL_INT_CHECK_GOTO(ret, < 0, err);
 
     type = (const struct http_client_node_type *)sol_flow_node_get_type(node);
-    if (streq(response->content_type, "application/json")) {
+
+    if (mdata->strict && mdata->accept && response->content_type &&
+        !streq(response->content_type, mdata->accept)) {
+        sol_flow_send_error_packet(node, EINVAL,
+            "Response has different content type. Received: %s - Desired: %s",
+            response->content_type, mdata->accept);
+        return;
+    }
+
+    if (streq(response->content_type, "application/json") &&
+        type->process_token) {
```

Comment:
why are you checking if `type->process_token` is not NULL and is not checking for `type->process_data` in the `else` statement ?

Also, is it expected the response be processed by `type->process_data` in case of this method not be defined ?